### PR TITLE
demo: update SGX images

### PR DESCRIPTION
**Pull request by mythi**
Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>

### CODE REVIEW

File: demo/sgx-aesmd-demo/Dockerfile [hunk 1]
```diff
@@ -2,9 +2,11 @@
...
 FROM ubuntu:20.04
 
-RUN apt update && apt install -y curl gnupg \
-    && echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main' | tee /etc/apt/sources.list.d/intel-sgx.list \
-    && curl -s https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - \
+RUN apt update && apt install -y curl gnupg-agent \
+    && echo "deb [arch=amd64 signed-by=/usr/share/keyrings/intel-sgx.gpg] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main" | \
+    tee -a /etc/apt/sources.list.d/intel-sgx.list \
+    && curl -s https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | \
+    gpg --dearmor --output /usr/share/keyrings/intel-sgx.gpg \
```
Comment (bart0sh) [inline demo/sgx-aesmd-demo/Dockerfile:9]: what's the difference between `apt-key add -` and `gpg --dearmor --output /usr/share/keyrings/intel-sgx.gpg ` ?

Comment (mythi) [reply demo/sgx-aesmd-demo/Dockerfile:9]: `apt-key` is deprecated.

**Comment by bart0sh** [comment]
@kad @rojkov PTAL. I don't have approval rights here.

**Review by rojkov — APPROVED** [review]

**Pull request merged**